### PR TITLE
[ENG-6539] Fixed archiving gv addons

### PR DESCRIPTION
**Pull request by opaduchak**
## QA Notes

Please make verification statements inspired by your code and what your code touches.
- Verify
- Verify

What are the areas of risk?

Any concerns/considerations/questions that development raised?

### CODE REVIEW

File: osf/external/gravy_valet/translations.py [hunk 1]
```diff
@@ -43,20 +47,14 @@ class _LegacyConfigsForWBKey(enum.Enum):
...
-        include_path = ('external_citation_service',)
-        service_key = gv_account_data.get_included_attribute(
-            include_path=include_path,
-            attribute_name='external_service_name'
-        )
+    addon_type = gv_account_data.resource_type.split('-')[1]
+    include_path = f'external_{gv_account_data.resource_type.split('-')[1]}_service',
+    service_wb_key = gv_account_data.get_included_attribute(
+        include_path=include_path,
+        attribute_name='external_service_name' if addon_type == AddonType.CITATION else 'wb_key'
```
Comment (opaduchak) [inline osf/external/gravy_valet/translations.py:54]: This is to consistent with current stage configuration. If `wb_key` for citation addons is configured properly, we may replace it with 
```suggestion
        attribute_name='wb_key'
```

File: osf/external/gravy_valet/translations.py [hunk 2]
```diff
@@ -65,26 +63,21 @@ def make_ephemeral_user_settings(gv_account_data, requesting_user):
...
-            include_path=include_path,
-            attribute_name='wb_key'
-        )
-    legacy_config = _LegacyConfigsForWBKey[service_key].value
-    return EphemeralNodeSettings(
+    addon_type = gv_addon_data.resource_type.split('-')[1]
+    include_path = ('base_account', f'external_{addon_type}_service')
+    service_wb_key = gv_addon_data.get_included_attribute(
+        include_path=include_path,
+        attribute_name='external_service_name' if addon_type == AddonType.CITATION else 'wb_key'
```
Comment (opaduchak) [inline osf/external/gravy_valet/translations.py:70]: This is to consistent with current stage configuration. If `wb_key` for citation addons is configured properly, we may replace it with
```suggestion
        attribute_name='wb_key'
```

File: osf/external/gravy_valet/request_helpers.py [hunk 3]
```diff
@@ -257,8 +257,10 @@ def get_gv_citation_url_list_for_project(auth, addon_short_name, project, reques
 def _invoke_gv_citation_operation_invocations(auth, addon, project, list_id):
     data = {
         'attributes': {
-            'operation_name': 'list_root_collections',
-            'operation_kwargs': {},
+            'operation_name': 'list_collection_items',
+            'operation_kwargs': {
+                'collection_id': addon['attributes']['root_folder'] if list_id == 'ROOT' else list_id
```
Comment (opaduchak) [inline osf/external/gravy_valet/request_helpers.py:262]: I've noticed the issue when actual root items would be fetched instead if child items of configured root. This resolves it
@adlius fyi

File: osf/external/gravy_valet/translations.py [hunk 4]
```diff
@@ -284,3 +282,29 @@ def after_fork(self, node: 'Node', fork, user: 'OSFUser', save=True):
...
+
+def _initialize_ephemeral_storage_node_settings():
+    from addons.base.models import BaseStorageAddon
+
+    global _StorageEphemeralNodeSettings
+
+    class StorageEphemeralNodeSettings(EphemeralNodeSettings, BaseStorageAddon):
+        pass
+
+    _StorageEphemeralNodeSettings = StorageEphemeralNodeSettings
```
Comment (opaduchak) [inline osf/external/gravy_valet/translations.py:307]: I'm not happy with this approach either, so I am more than welcome to suggestions

Comment (brianjgeiger) [reply osf/external/gravy_valet/translations.py:307]: I'm afraid I don't have any other suggestions.

File: osf/external/gravy_valet/translations.py (hunk 1 above)
Comment (brianjgeiger) [reply osf/external/gravy_valet/translations.py:54]: Will we also need to do this for computing addons?

**Review by brianjgeiger — APPROVED** [review]

File: osf/external/gravy_valet/translations.py (hunk 1 above)
Comment (opaduchak) [reply osf/external/gravy_valet/translations.py:54]: If we set `wb_key` in GV admin, we won't

Comment (adlius) [reply osf/external/gravy_valet/translations.py:54]: I have two PRs that fix this problem. In the [first one](https://github.com/CenterForOpenScience/gravyvalet/pull/178),  is I added `external_service_name` to the GV ExternalService serializer. And I changed `make_ephemeral_user_settings` and `make_ephemeral_node_settings` in [this PR](https://github.com/CenterForOpenScience/osf.io/pull/10837/files).

**Review by adlius — COMMENTED** [review]
I have some other PRs that I made that fixed some of the issues. I think you might want to take a look at those:
https://github.com/CenterForOpenScience/gravyvalet/pull/178/files
https://github.com/CenterForOpenScience/osf.io/pull/10837/files

File: osf/external/gravy_valet/translations.py (hunk 4 above)
Comment (Johnetordoff) [reply osf/external/gravy_valet/translations.py:307]: Might avoid using a global with this:
```
def _initialize_ephemeral_storage_node_settings():
    if not hasattr(_initialize_ephemeral_storage_node_settings, "_storage_node_settings"):
        from addons.base.models import BaseStorageAddon

        class StorageEphemeralNodeSettings(EphemeralNodeSettings, BaseStorageAddon):
            pass

        _initialize_ephemeral_storage_node_settings._storage_node_settings = StorageEphemeralNodeSettings

    return _initialize_ephemeral_storage_node_settings._storage_node_settings

```
I think that cover any import problems with a global.

**Review by brianjgeiger — APPROVED** [review]

**Pull request merged**